File: javascript/matriz_geral.js
const n = parseInt(prompt("Qual a ordem da matriz?"));

let matriz = [], somaPositivos = 0;

for(i = 0; i < n; i++) {
    matriz.push([]);
}

for(i = 0; i < n; i++) {
    for(j = 0; j < n; j++) {
        matriz[i][j] = parseFloat(prompt("Elemento [" + i + "," + j + "]:"));

        if(matriz[i][j] > 0) {
            somaPositivos += matriz[i][j];
        }
    }
}

console.log("SOMA DOS POSITIVOS: %.1f", somaPositivos);

let linha = parseInt(prompt("Escolha uma linha:"));

console.log("LINHA ESCOLHIDA: ");
let linhaEscolhida = "";
for(i = 0; i < n; i++) {
    linhaEscolhida += matriz[linha][i] + " ";
}
console.log(linhaEscolhida);


let coluna = parseInt(prompt("Escolha uma coluna:"));

console.log("COLUNA ESCOLHIDA: ");
let colunaEscolhida = "";
for(i = 0; i < n; i++) {
    colunaEscolhida += matriz[i][coluna] + " ";
}
console.log(colunaEscolhida);

console.log("DIAGONAL PRINCIPAL: ");
let diagonalPrincipal = "";
for(i = 0; i < n; i++) {
    for(j = 0; j < n; j++) {
        if(i == j) {
            diagonalPrincipal += matriz[i][j] + " ";
        }
    }
}
console.log(diagonalPrincipal);

console.log("MATRIZ ALTERADA: ");

for(i = 0; i < n; i++) {
    for(j = 0; j < n; j++) {
        if(matriz[i][j] < 0) {
            matriz[i][j] *= matriz[i][j];
        }
    }
}

for(i = 0; i < n; i++) {
    let aux = "";
    for(j = 0; j < n; j++) {
        aux += matriz[i][j] + " ";
    }
    console.log(aux);
}

File: python/matriz_geral.py
def criar_matriz(num_linhas: int, num_colunas: int) -> list[list[float]]:
    m = []

    for i in range(num_linhas):
        m.append([0] * num_colunas)

    return m


def ler_matriz(matriz: list[list[float]]) -> None:
    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            matriz[i][j] = float(input(f"Elemento [{i},{j}]: "))


def imprimir_matriz(matriz: list[list[float]]) -> None:
    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            print(matriz[i][j], end=" ")
        print()


n = int(input("Qual a ordem da matriz? "))
matriz = criar_matriz(n, n)
ler_matriz(matriz)

soma_positivos = 0

for i in range(n):
    for j in range(n):
        if matriz[i][j] > 0:
            soma_positivos += matriz[i][j]

print(f"SOMA DOS POSITIVOS: {soma_positivos:.1f}")

linha = int(input("Escolha uma linha: "))
print("LINHA ESCOLHIDA: ", end="")
for i in range(n):
    print(f"{matriz[linha][i]:.1f}", end=" ")

print()

coluna = int(input("Escolha uma coluna: "))
print("COLUNA ESCOLHIDA: ", end="")
for i in range(n):
    print(f"{matriz[i][coluna]:.1f}", end=" ")

print()

print("DIAGONAL PRINCIPAL: ", end="")
for i in range(n):
    print(f"{matriz[i][i]:.1f}", end=" ")

print()

matriz_alterada = criar_matriz(n, n)
for i in range(n):
    for j in range(n):
        matriz_alterada[i][j] = matriz[i][j]

        if matriz_alterada[i][j] < 0:
            matriz_alterada[i][j] = matriz_alterada[i][j] ** 2

print("MATRIZ ALTERADA:")
imprimir_matriz(matriz_alterada)
